picasso/src/main/java/com/squareup/picasso/RequestCreator.java

```diff
@@ -373,8 +378,19 @@ public void fetch() {
 
       Request request = createRequest(started);
       String key = createKey(request, new StringBuilder());
+      Bitmap bitmap = picasso.quickMemoryCacheCheck(key);
```

Comment:
I don't think thread safety really matters here anyway. Worst case you cache miss and just do the fetch which is what you'd do otherwise.